pkg/telemetry/storage/db.go

```diff
@@ -76,13 +76,14 @@ func (db *TelemetryDatabase) StoreMetric(ctx context.Context, metric telemetry.M
 	return nil
 }
 
-// FetchUnpublishedMetrics fetches unplished metrics.
-func (db *TelemetryDatabase) FetchUnpublishedMetrics(ctx context.Context, amount int) ([]telemetry.Metric, error) {
+// FetchMetrics fetches unplished metrics.
+func (db *TelemetryDatabase) FetchMetrics(ctx context.Context, published bool, amount int) ([]telemetry.Metric, error) {
 	rows, err := db.sqlDB.QueryContext(ctx,
 		`SELECT rowid, version, timestamp, type, payload, published FROM system_metrics 
-		WHERE published is false 
+		WHERE published is ?1 
 		ORDER BY timestamp
-		LIMIT ?1`,
+		LIMIT ?2`,
+		published,
 		amount,
 	)
```

Comment:
Q: Makes sense to move this and other raw sqls in this file to sqlc?

Reply:
hum. this is a different database. I completely forgot about this when putting the `sqlc` back. I only put it in the main database. I think we should use `sqlc` here too. But probably do it in another ticket.